homeassistant/components/inels/switch.py

```diff
@@ -0,0 +1,180 @@
+                    entities.append(
+                        InelsSwitch(
+                            device=device,
+                            description=entity_description,
+                            index=idx,
+                        )
+                    )
+
+    async_add_entities(entities, False)
+
+
+class InelsSwitch(InelsBaseEntity, SwitchEntity):
+    """The platform class required by Home Assistant."""
+
+    entity_description: InelsSwitchEntityDescription
+
+    def __init__(
+        self,
+        device: Device,
+        description: InelsSwitchEntityDescription,
+        index: int = 0,
+    ) -> None:
+        """Initialize the switch."""
+        super().__init__(device=device, key=description.key, index=index)
+        self.entity_description = description
+
+        # Include index in unique_id for devices with multiple switches
+        unique_key = f"{description.key}{index}" if index else description.key
+
+        self._attr_unique_id = slugify(f"{self._attr_unique_id}_{unique_key}")
+        self._attr_name = description.name
+
+    @property
+    def available(self) -> bool:
+        """Return entity availability."""
+        if self.entity_description.alerts:
+            try:
+                last_state = self.entity_description.get_last_state_fn(
+                    self._device, self._index
+                )
+            except (KeyError, IndexError, AttributeError):
+                last_state = None
+
+            current_value = self.entity_description.get_state_fn(
+                self._device, self._index
+            )
+            for alert_key, alert_msg in self.entity_description.alerts:
+                if getattr(current_value, alert_key, None):
+                    if not last_state or not getattr(last_state, alert_key, None):
+                        LOGGER.warning(alert_msg, self.name, self._device.state_topic)
+                    return False
+        return super().available
+
+    @property
+    def icon(self) -> str | None:
+        """Switch icon."""
+        return self.entity_description.icon
+
+    @callback
+    def _async_update_attrs(self) -> None:
+        """Update attrs from device."""
+        self._attr_is_on = self.entity_description.value_fn(self._device, self._index)
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return if switch is on."""
+        return self.entity_description.value_fn(self._device, self._index)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Instruct the switch to turn off."""
+        if not self._device.is_available:
+            return
+        self.entity_description.set_fn(self._device, self._index, False)
+        await self.hass.async_add_executor_job(
+            self._device.set_ha_value, self._device.state
+        )
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Instruct the switch to turn on."""
+        if not self._device.is_available:
+            return
```

Comment:
The default implementation of the entity is that it goes unavailable once `is_available` is False, so then this could not be executed right?